### PR TITLE
feat: web-sdk native support

### PR DESCRIPTION
**Pull request by samialdury**
`RELAY_PROXY_HOST` env variable isn't currently used anywhere, should we remove this option?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,10 +8,12 @@ by your visitors.
...
-| ------------------ | -------- | ------------- | ------------------------------------------------------------------- |
-| `RELAY_PROXY_HOST` | `string` | `null`        | Hostname on which will the proxy run on                             |
-| `LOGGER_LEVEL`     | `string` | `'info'`      | One of `trace`, `debug`, `info`, `warn`, `error`, `fatal`, `silent` |
+| Name                      | Type     | Default value                             | Description                                                         |
+| ------------------------- | -------- | ----------------------------------------- | ------------------------------------------------------------------- |
+| `RELAY_PROXY_HOST`        | `string` | `null`                                    | Hostname on which will the proxy run on                             |
+| `LOGGER_LEVEL`            | `string` | `'info'`                                  | One of `trace`, `debug`, `info`, `warn`, `error`, `fatal`, `silent` |
+| `MANAGER_HOST`            | `string` | `'https://manager.eu.smartlook.com'`      | Smartlook Manager host                                              |
```
Comment (Pajk) [inline README.md:15]: Please set the default to https://manager.eu.smartlook.cloud.

File: package.json [hunk 2]
```diff
@@ -22,24 +22,24 @@
 	},
 	"dependencies": {
 		"convict": "^6.2.1",
-		"pino": "^7.6.2",
-		"undici": "^4.12.1"
+		"pino": "^7.8.0",
```
Comment (Pajk) [inline package.json:25]: You can configure dependabot to keep the dependencies always up to date.

File: src/config/index.ts [hunk 3]
```diff
@@ -36,20 +36,20 @@ export const config = convict({
...
-				env: 'MOBILE_SDK_HOST',
+				env: 'MOBILE_SDK_WRITER_HOST',
 				default: 'https://sdk-writer.eu.smartlook.cloud',
 			},
 			manager: {
 				doc: 'Manager host',
 				format: String,
 				env: 'MANAGER_HOST',
-				default: 'https://manager.smartlook.com',
+				default: 'https://manager.eu.smartlook.com',
```
Comment (Pajk) [inline src/config/index.ts:52]: .cloud

File: src/proxy/routes/index.ts [hunk 4]
```diff
@@ -1,124 +1,36 @@
...
+		prefix: '/web-writer',
+		stripPrefix: true,
+	},
+	// {
+	// 	name: 'sdkWriter',
+	// 	targetHost: config.get('proxy.hosts.mobileSdkWriter'),
+	// 	prefix: '/sdk-writer',
+	// 	stripPrefix: true,
+	// },
+	// Request without prefix -> web-sdk
```
Comment (Pajk) [inline src/proxy/routes/index.ts:29]: Why is this needed?

File: src/proxy/utils.ts [hunk 5]
```diff
@@ -1,106 +1,42 @@
...
 ): IncomingHttpHeaders => {
+	const webWriterHost = config
+		.get('proxy.hosts.webSdkWriter')
+		.split('https://')[1]
+
+	const host = url.includes('/rec/v2/write')
+		? webWriterHost
+		: // Next line bypasses: [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames.
+		  ''
+
```
Comment (Pajk) [inline src/proxy/utils.ts:31]: I would rather propagate the host header from the "routes" config and set it for all services.

**Review by Pajk — COMMENTED** [review]
Yes, let's get rid of RELAY_PROXY_HOST if we don't need it anymore.

File: src/proxy/utils.ts [hunk 6]
```diff
@@ -1,125 +1,53 @@
...
-	}
-}
-
-export const buildUrl = (route: IProxyRouteConfig, url: string): string => {
-	return `${route.targetHost}${stripPrefix(url, route.proxyPrefix)}`
-}
+): IncomingHttpHeaders => ({
+	...originalHeaders,
+	'x-forwarded-for': originalHeaders['x-forwarded-for'] ?? remoteAddress,
+	'x-forwarded-host': host,
```
Comment (Pajk) [inline src/proxy/utils.ts:24]: Is `x-forwarded-host` needed if you always set the `host` header?

File: src/proxy/utils.ts [hunk 7]
```diff
@@ -1,125 +1,53 @@
...
+	host,
+})
 
 export const pipeResponse = async (
+	routeTargetHost: string,
 	url: string,
 	req: IncomingMessage,
 	res: ServerResponse
 ): Promise<void> => {
+	const host = routeTargetHost.split('https://')[1] ?? ''
```
Comment (Pajk) [inline src/proxy/utils.ts:34]: I would prefer `routeTargetHost.replace('https://', '')`. It just create one string instead of an array with two strings.

**Review by Pajk — APPROVED** [review]

File: src/proxy/utils.ts (hunk 7 above)
Comment (samialdury) [reply src/proxy/utils.ts:34]: Good point👍🏽

File: src/proxy/utils.ts (hunk 6 above)
Comment (samialdury) [reply src/proxy/utils.ts:24]: https://github.com/koajs/koa/blob/master/docs/api/request.md#requesthostname
